packages/core/src/server/service.ts

```diff
@@ -1,47 +1,59 @@
 import http from "node:http";
 import type { Common } from "@/common/common.js";
-import type { MetadataStore, ReadonlyStore } from "@/indexing-store/store.js";
-import { graphiQLHtml } from "@/ui/graphiql.html.js";
+import type { DatabaseService } from "@/database/service.js";
+import { convertSchemaToDrizzle, createDrizzleDb } from "@/drizzle/runtime.js";
+import { graphql } from "@/graphql/index.js";
+import { type PonderRoutes, applyHonoRoutes } from "@/hono/index.js";
+import { getMetadataStore } from "@/indexing-store/metadata.js";
+import { getReadonlyStore } from "@/indexing-store/readonly.js";
+import type { Schema } from "@/schema/common.js";
 import { startClock } from "@/utils/timer.js";
-import { maxAliasesPlugin } from "@escape.tech/graphql-armor-max-aliases";
-import { maxDepthPlugin } from "@escape.tech/graphql-armor-max-depth";
-import { maxTokensPlugin } from "@escape.tech/graphql-armor-max-tokens";
 import { serve } from "@hono/node-server";
-import { GraphQLError, type GraphQLSchema } from "graphql";
-import { createYoga } from "graphql-yoga";
 import { Hono } from "hono";
 import { cors } from "hono/cors";
 import { createMiddleware } from "hono/factory";
 import { createHttpTerminator } from "http-terminator";
-import {
-  type GetLoader,
-  buildLoaderCache,
-} from "./graphql/buildLoaderCache.js";
+import { onError } from "./error.js";
 
 type Server = {
-  hono: Hono<{ Variables: { store: ReadonlyStore; getLoader: GetLoader } }>;
+  hono: Hono;
   port: number;
   kill: () => Promise<void>;
 };
 
 export async function createServer({
-  graphqlSchema,
-  readonlyStore,
-  metadataStore,
+  app: userApp,
+  routes: userRoutes,
   common,
+  schema,
+  database,
+  dbNamespace,
 }: {
-  graphqlSchema: GraphQLSchema;
-  readonlyStore: ReadonlyStore;
-  metadataStore: MetadataStore;
+  app?: Hono;
+  routes?: PonderRoutes;
   common: Common;
+  schema: Schema;
+  database: DatabaseService;
+  dbNamespace: string;
 }): Promise<Server> {
-  const hono = new Hono<{
-    Variables: { store: ReadonlyStore; getLoader: GetLoader };
-  }>();
+  // Create hono app
 
-  let port = common.options.port;
   const startTime = Date.now();
 
+  const readonlyStore = getReadonlyStore({
```

Comment:
Why'd you choose to instantiate the stores here rather than in `bin`?

Reply:
Seemed more concise, don't have to have duplicate code in `bin/commands/serve.ts` and `bin/utils/runServer.ts`